File: day02/R/main.R
library(data.table)
dt <- fread('input.txt')
dt[,
   .(forward = sum((V1 == 'forward') * V2),
     depth = sum((V1 == 'down') * V2 - (V1 == 'up') * V2))][,
       .(part1 = forward * depth)
     ]

dt[,
   .(forward = (V1 == 'forward') * V2,
     aim = cumsum((V1 == 'down') * V2 - (V1 == 'up') * V2))][
       ,
       .(part2 = sum(forward) * sum(forward * aim))
       ]

File: day02/python/main.py
import numpy as np

arr = [l.strip() for l in open('input.txt')]

x = 0
y = 0

print(arr)
for l in arr:
    l = l.split()
    print(l)
    if l[0] == 'forward':
        x += int(l[1])
    elif l[0] == 'down':
        y += int(l[1])
    elif l[0] == 'up':
        y -= int(l[1])

print('Part 1:', x*y)


arr = [l.strip() for l in open('input.txt')]

x = 0
y = 0
aim = 0

print(arr)
for l in arr:
    l = l.split()
    print(l)
    if l[0] == 'forward':
        x += int(l[1])
        y += aim * int(l[1])
    elif l[0] == 'down':
        aim += int(l[1])
    elif l[0] == 'up':
        aim -= int(l[1])

print('Part 2:', x*y)